### PR TITLE
chore: Remove `register-finality-provider` cmd

### LINKED ISSUE #188: Remove `register-finality-provider` and use `create-finality-provider` for registration
Currently, user needs to perform `create-finality-provider` to generate all the data needed for registration before running `register-finality-provider`, which is not necessary. User should be able to register fp by only running `create-finality-provider`. This also follows Cosmos SDK's convention of using `create-validator` for registration

### PR DESCRIPTION
**Pull request by gitferry**
Closes #188. In particular, this PR:
- removed `register-finality-provider` cmd
- move functionalities of registration to `create-finality-provider`
- removed status of `CREATED`
- fixed tests
- updated docs
- some cleanups (e.g., moved event loops from `app.go` to `event_loops.go`)

As this PR caused major change in operations, please give it a careful look

### CODE REVIEW

**Review by Lazar955 — APPROVED** [review]
great job! love the `-1,246` lines removed

File: finality-provider/service/app.go [hunk 1]
```diff
@@ -351,31 +294,79 @@ func (app *FinalityProviderApp) Stop() error {
...
+	// 2. create proof-of-possession
+	if eotsPk == nil {
+		return nil, fmt.Errorf("eots pk cannot be nil")
+	}
+	pop, err := app.CreatePop(fpAddr, eotsPk, passPhrase)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create proof-of-possession of the finality-provider: %w", err)
+	}
+
+	// TODO: query consumer chain to check if the fp is already registered
```
Comment (RafilxTenfen) [inline finality-provider/service/app.go:323]: good point

File: finality-provider/proto/finality_providers.proto [hunk 2]
```diff
@@ -58,37 +53,25 @@ message CreateFinalityProviderRequest {
     string key_name = 1;
     // passphrase is used to encrypt the keys
     string passphrase = 2;
-    // hd_path is the hd path for private key derivation
```
[the remark this answers is not in the document]
Comment (gitferry) [reply finality-provider/proto/finality_providers.proto:61]: In https://github.com/babylonlabs-io/finality-provider/pull/168 we disallow creation of keys in `create-finality-provider`, so `hd-path` is not needed

**Review by RafilxTenfen — APPROVED** [review]
utACK, we should check this version of fpd in bbn-deployments

**Pull request merged**